numeric.go

```diff
@@ -448,8 +448,11 @@ func (src *Numeric) toFloat64() (float64, error) {
 		return math.Inf(-1), nil
 	}
 
-	if src.Exp == 1 {
+	switch {
```

Comment:
This optimization isn't present in pgx@v5 so I don't think there's anything to fix in v5.

https://github.com/jackc/pgx/blob/20bf953a17fc7cf6619536a87bd285b9273dbcf9/pgtype/numeric.go#L66

Reply:
Please double-check my logic. I think src.Int is normalized, so this optimization is valid.

Reply:
I think there is a danger of integer overflow in `float64(src.Int.Int64() * 10)`. I think it should be `float64(src.Int.Int64()) * 10` instead. But more so, is that branch even necessary / valuable. I assume it would work without it.

Reply:
I removed the case for `exp == 1` since you're right; it's not necessary. I kept that case since I thought that might have been the intent of the original code.

If you wanted to get fancy, you could implement it for all positive exponents, but I'll leave just the particular case for exp == 0 for now.

``` go
	if src.Exp >= 0 {
		return float64(src.Int.Int64()) * math.Pow10(int(src.Exp)), nil
	}
```